feat(installer/windows): add wix installer

ReenigneArcher opened the pull request:
## Description
<!--- Please include a summary of the changes. --->
This PR adds a WIX installer for Sunshine. The NSIS installer will be removed at a later point after any bugs with the WIX installer are ironed out.

This is a requirement for having an ARM64 installer for https://github.com/LizardByte/Sunshine/pull/3905 or https://github.com/LizardByte/Sunshine/pull/3369

TODO:
- [x] custom actions are not running
- [x] shortcut is not created
- [x] vigembus installer won't run because we're already running an msi installer... proposed solution: https://github.com/LizardByte/Sunshine/pull/4625
- [x] application icon not showing in installed apps screen
- [x] docs for creating installer (need dotnet)
- [x] fix start menu directory structure
- [x] should we hide cmd/powershell windows? https://github.com/LizardByte/Sunshine/pull/4675
- [x] document process for migrating from old installer to new one

For first release after this is merged, we will keep both exe and msi, so that the winget automation will still work to auto update winget. We will then to update winget manually to use the msi
- https://github.com/vedantmgoyal9/winget-releaser/issues/322
- https://github.com/microsoft/winget-pkgs/tree/master/manifests/l/LizardByte/Sunshine
- [ ] change winget from exe to msi

### Screenshot
<!--- Include screenshots if the changes are UI-related. --->


### Issues Fixed or Closed
<!--- Close issue example: `- Closes #1` --->
<!--- Fix bug issue example: `- Fixes #2` --->
<!--- Resolve issue example: `- Resolves #3` --->
- Resolves https://github.com/LizardByte/roadmap/issues/76


## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [x] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

cgutman commented:
The latest NSIS version now runs properly on ARM64 systems, so we can stick with NSIS if we want. WiX does make it easier to support having a single unified installer for both x64 and ARM64 binaries if we want that.

ReenigneArcher commented:
Good to know. I still need to setup a VM to test this out.

I think msi would still be nice to use, especially for anyone installing Sunshine with a script as it supports more CLI options. I discovered in a recent change that I couldn't even get the output of the installer scripts to a log file with NSIS.

Do you know the process for unifying the installer? That would be pretty nice.

ReenigneArcher committed:
feat(installer/windows): add wix installer

rbqvq commented:
It seems have some problem

```
"C:/sunshine/build/.wix/wix.exe" build -arch arm64 -out "C:/sunshine/build/cpack_artifacts/_CPack_Packages/win64/WIX/Sunshine.msi" -ext "WixToolset.UI.wixext" -ext "WixToolset.Util.wixext" -src "C:/sunshine/build/cpack_artifacts/_CPack_Packages/win64/WIX/directories.wxs" -src "C:/sunshine/build/cpack_artifacts/_CPack_Packages/win64/WIX/files.wxs" -src "C:/sunshine/build/cpack_artifacts/_CPack_Packages/win64/WIX/features.wxs" -src "C:/sunshine/build/cpack_artifacts/_CPack_Packages/win64/WIX/main.wxs" -src "C:/sunshine/build/wix_packaging/sunshine-installer.wxs" -i "C:/sunshine/build/cpack_artifacts/_CPack_Packages/win64/WIX"
wix.exe : error WIX0144: The extension 'WixToolset.UI.wixext' could not be found. Checked paths: WixToolset.UI.wixext
```

I'm try resolve it but I don't knows the reasons, manual add extension still failed :(

rbqvq commented:
Oh, that might be my environment issue.

I know, Thanks.

ReenigneArcher commented:
> Should we wpdate getting_started.md installer links?
> 
> Provide MSI links only or both EXE and MSI?

PR is draft, review not requested. I already have updating docs on the todo in the PR request body.

ReenigneArcher commented:
Simplified installer by adding a new powershell script in https://github.com/LizardByte/Sunshine/pull/4675 and refactored this PR to use it. powershell window is hidden when using quiet install/uninstall and shown otherwise. The script has decent feedback so I don't think it's an issue to show it, but we can easily hide it if we want. The output is also sent to a log file to help with debugging installation issues.
